comp/metadata/securityagent/impl/security_agent.go

```diff
@@ -0,0 +1,182 @@
+	type PayloadAlias Payload
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload implements marshaler.AbstractMarshaler#SplitPayload.
+//
+// In this case, the payload can't be split any further.
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("could not split security-agent process payload any more, payload is too big for intake")
+}
+
+type securityagent struct {
+	util.InventoryPayload
+
+	log  log.Component
+	conf config.Component
+	// sysprobeConf optional.Option[sysprobeconfig.Component]
+	hostname string
+}
+
+// Requires defines the dependencies for the securityagent metadata component
+type Requires struct {
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+	// We need the authtoken to be created so we requires the comp. It will be used by configFetcher.
+	AuthToken authtoken.Component
+	// SysProbeConfig optional.Option[sysprobeconfig.Component]
+}
+
+// Provides defines the output of the securityagent metadata component
+type Provides struct {
+	Comp             def.Component
+	MetadataProvider runnerimpl.Provider
+	FlareProvider    flaretypes.Provider
+	Endpoint         api.AgentEndpointProvider
+}
+
+// NewComponent creates a new securityagent metadata Component
+func NewComponent(deps Requires) Provides {
+	hname, _ := hostname.Get(context.Background())
+	sa := &securityagent{
+		log:      deps.Log,
+		conf:     deps.Config,
+		hostname: hname,
+	}
+	sa.InventoryPayload = util.CreateInventoryPayload(deps.Config, deps.Log, deps.Serializer, sa.getPayload, "security-agent.json")
+
+	return Provides{
+		Comp:             sa,
+		MetadataProvider: sa.MetadataProvider(),
+		FlareProvider:    sa.FlareProvider(),
+		Endpoint:         api.NewAgentEndpointProvider(sa.writePayloadAsJSON, "/metadata/security-agent", "GET"),
+	}
+}
+
+func (sa *securityagent) writePayloadAsJSON(w http.ResponseWriter, _ *http.Request) {
+	// GetAsJSON calls getPayload which already scrub the data
+	scrubbed, err := sa.GetAsJSON()
+	if err != nil {
+		utils.SetJSONError(w, err, 500)
+		return
+	}
+	w.Write(scrubbed)
+}
+
+func (sa *securityagent) getConfigLayers() map[string]interface{} {
+	metadata := map[string]interface{}{
+		"agent_version": version.AgentVersion,
+	}
+
+	// TODO: ??? do we use this
+	if !sa.conf.GetBool("inventories_configuration_enabled") {
+		return metadata
+	}
+
+	// FIXME: we should use a subset of the config from the security-agent
+	if sa.conf == nil {
+		sa.log.Debugf("security-agent config not available: disabling securityagent metadata")
+		return metadata
+	}
+
+	// FIXME: we should use a subset of the config from the security-agent
+	rawLayers, err := fetchSecurityAgentConfigBySource(sa.conf)
+	if err != nil {
+		sa.log.Debugf("error fetching security-agent config layers: %s", err)
+		return metadata
+	}
+
+	configBySources := map[string]interface{}{}
+	fmt.Println("rawLayers")
```

Comment:
Left over from debug I think

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/26160/commits/b75328d0788ef1aaf102214124717a445c403d81